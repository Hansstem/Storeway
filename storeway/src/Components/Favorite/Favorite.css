.heartIcon {
	color: #ff0909;
	font-size: 2rem;
	cursor: pointer;
	z-index: 3;
}
