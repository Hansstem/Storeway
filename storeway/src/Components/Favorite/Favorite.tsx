import './Favorite.css'
import { useEffect, useState } from 'react'
import IonIcon from '@reacticons/ionicons'

interface MyProp {
	handleClick: () => void
	id: string
}
export default function Favorite(props: MyProp) {
	const [isClick, setClick] = useState(false)
	function click() {
		setClick(!isClick)
		props.handleClick()
	}

	useEffect(() => {
		const favorites = localStorage.getItem('FavoriteList')
		const favoriteList = favorites?.split(',')
		if (favoriteList) {
			favoriteList.forEach(e => {
				if (e === props.id.toString()) {
					setClick(true)
				}
			})
		}
	}, [props.id])

	return (
		<IonIcon
			name={isClick ? 'heart' : 'heart-outline'}
			onClick={() => click()}
			className="heartIcon"
		/>
	)
}
