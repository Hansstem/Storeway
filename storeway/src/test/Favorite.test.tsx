import { render, fireEvent } from '@testing-library/react'
import Favorite from '../Components/Favorite/Favorite'
import jest from 'jest-mock'

describe('Heart icon test', () => {
	it('State test when clicking on heart icon', () => {
		const handleClick = jest.fn() // Mock handleClick function
		const { getByRole } = render(<Favorite handleClick={handleClick} id="1" />)
		expect(getByRole('heartIcon')).toHaveAttribute('name', 'heart-outline')
		fireEvent.click(getByRole('heartIcon'))
		expect(getByRole('heartIcon')).toHaveAttribute('name', 'heart')
		expect(handleClick).toHaveBeenCalled()
		fireEvent.click(getByRole('heartIcon'))
		expect(getByRole('heartIcon')).toHaveAttribute('name', 'heart-outline')
	})
})
