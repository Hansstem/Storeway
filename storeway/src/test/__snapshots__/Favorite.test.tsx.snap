// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Heart icon test > State test when clicking on heart icon 1`] = `
<DocumentFragment>
  <span
    class="heartIcon"
    name="heart-outline"
    role="heartIcon"
    style="display: inline-block; height: 1em; width: 1em; visibility: inherit; fill: currentcolor; box-sizing: content-box; stroke: currentcolor;"
  >
    <svg
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M352.92,80C288,80,256,144,256,144s-32-64-96.92-64C106.32,80,64.54,124.14,64,176.81c-1.1,109.33,86.73,187.08,183,252.42a16,16,0,0,0,18,0c96.26-65.34,184.09-143.09,183-252.42C447.46,124.14,405.68,80,352.92,80Z"
        style="fill:none;stroke:currentColor;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Heart icon test > State test when clicking on heart icon 2`] = `
<DocumentFragment>
  <span
    class="heartIcon"
    name="heart"
    role="heartIcon"
    style="display: inline-block; height: 1em; width: 1em; visibility: inherit; fill: currentcolor; box-sizing: content-box; stroke: currentcolor;"
  >
    <svg
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,448a32,32,0,0,1-18-5.57c-78.59-53.35-112.62-89.93-131.39-112.8-40-48.75-59.15-98.8-58.61-153C48.63,114.52,98.46,64,159.08,64c44.08,0,74.61,24.83,92.39,45.51a6,6,0,0,0,9.06,0C278.31,88.81,308.84,64,352.92,64,413.54,64,463.37,114.52,464,176.64c.54,54.21-18.63,104.26-58.61,153-18.77,22.87-52.8,59.45-131.39,112.8A32,32,0,0,1,256,448Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Tester state n책r man trykker p책 hjertet 1`] = `
<DocumentFragment>
  <span
    class="heartIcon"
    name="heart-outline"
    role="heartIcon"
    style="display: inline-block; height: 1em; width: 1em; visibility: inherit; fill: currentcolor; box-sizing: content-box; stroke: currentcolor;"
  >
    <svg
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M352.92,80C288,80,256,144,256,144s-32-64-96.92-64C106.32,80,64.54,124.14,64,176.81c-1.1,109.33,86.73,187.08,183,252.42a16,16,0,0,0,18,0c96.26-65.34,184.09-143.09,183-252.42C447.46,124.14,405.68,80,352.92,80Z"
        style="fill:none;stroke:currentColor;stroke-linecap:round;stroke-linejoin:round;stroke-width:32px"
      />
    </svg>
  </span>
</DocumentFragment>
`;

exports[`Tester state n책r man trykker p책 hjertet 2`] = `
<DocumentFragment>
  <span
    class="heartIcon"
    name="heart"
    role="heartIcon"
    style="display: inline-block; height: 1em; width: 1em; visibility: inherit; fill: currentcolor; box-sizing: content-box; stroke: currentcolor;"
  >
    <svg
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256,448a32,32,0,0,1-18-5.57c-78.59-53.35-112.62-89.93-131.39-112.8-40-48.75-59.15-98.8-58.61-153C48.63,114.52,98.46,64,159.08,64c44.08,0,74.61,24.83,92.39,45.51a6,6,0,0,0,9.06,0C278.31,88.81,308.84,64,352.92,64,413.54,64,463.37,114.52,464,176.64c.54,54.21-18.63,104.26-58.61,153-18.77,22.87-52.8,59.45-131.39,112.8A32,32,0,0,1,256,448Z"
      />
    </svg>
  </span>
</DocumentFragment>
`;
